bunsen/bunsen-core/src/main/java/com/cerner/bunsen/definitions/StructureDefinitions.java

```diff
@@ -53,17 +66,366 @@ public abstract class StructureDefinitions {
+        childFields.addAll(childField);
+      }
+
+      T result = visitor.visitParentExtension(sliceName,
+          url,
+          childFields);
+
+      if (result == null) {
+        return Collections.emptyList();
+      } else {
+        return Collections.singletonList(
+            StructureField.extension(sliceName,
+                url,
+                extensionRoot.getIsModifier(),
+                result));
+      }
+    } else {
+      // The extension has no children, so produce its value.
+      Optional<IElementDefinition> valueElement = children.stream()
+          .filter(e -> e.getPath().contains("value"))
+          .findFirst();
+      // FIXME: get the extension URL.
+      Optional<IElementDefinition> urlElement = children.stream()
+          .filter(e -> e.getPath().endsWith("url"))
+          .findFirst();
+      String extensionUrl = urlElement.get().getFixedPrimitiveValue();
+      List<StructureField<T>> childField = elementToFields(visitor, rootDefinition,
+          valueElement.get(), extensionDefinitions, stack);
+      T result = visitor.visitLeafExtension(sliceName,
+          extensionUrl,
+          childField.iterator().next().result());
+      return Collections.singletonList(
+          StructureField.extension(sliceName,
+              extensionUrl,
+              extensionRoot.getIsModifier(),
+              result));
+    }
+  }
+
+  /**
+   * Returns the fields for the given element. The returned stream can be empty
+   * (e.g., for elements with max of zero), or have multiple values (for elements
+   * that generate fields with additional data in siblings.)
+   */
+  private <T> List<StructureField<T>> elementToFields(DefinitionVisitor<T> visitor,
+      IStructureDefinition rootDefinition,
+      IElementDefinition element,
+      List<IElementDefinition> snapshotDefinitions,
+      Deque<QualifiedPath> stack) {
+
+    String elementName = DefinitionVisitorsUtil.elementName(element.getPath());
+
+    if (element.getMax().equals("0")) {
+      // Fields with max of zero are omitted.
+      return Collections.emptyList();
+    } else if ("Extension".equals(element.getFirstTypeCode())) {
+      return extensionElementToFields(visitor, rootDefinition, element, snapshotDefinitions, stack);
+    } else if (element.getSliceName() != null) {
+      // Drop slices for non-extension fields; otherwise we will end up with duplicated fields.
+      return Collections.emptyList();
+    } else if (element.hasSingleType() && PRIMITIVE_TYPES.contains(element.getFirstTypeCode())) {
+      T primitiveConverter = visitor
+          .visitPrimitive(elementName, element.getFirstTypeCode());
+      if (!element.getMax().equals("1")) {
+        return singleField(elementName, visitor.visitMultiValued(elementName, primitiveConverter));
+      } else {
+        return singleField(elementName, primitiveConverter);
+      }
+
+    } else if (element.getPath().endsWith("[x]") && !element.getPath().startsWith("Extension")) {
```

Comment:
The conditional check for `!element.getPath().startsWith("Extension")` was earlier present only for `R4StructureDefintions` and now it is generalised for all. Hope this is also fine

Reply:
Yeah for this too I think the actual bug was triggered for R4 version of US-Core profile but it might be generic. Regardless, according to notes in #559 (and referenced issues like #286) having this check should be fine for STU3 as well as I don't think we properly handle `[x]` types for extensions right now.